test/e2e/setup-e2e.sh

```diff
@@ -52,17 +51,16 @@ if [ ! -z ${SYSTEM_MSI_CLUSTER:-} ]; then
   fi
 fi
 
-[[ ! -z "${AZURE_CLIENT_ID:-}" ]]         || (echo 'Must specify AZURE_CLIENT_ID' && exit -1)
+[[ ! -z "${AZURE_CLIENT_ID:-}" ]] || (echo 'Must specify AZURE_CLIENT_ID' && exit -1)
 
 # Create a keyvault
 az keyvault create -g "$RESOURCE_GROUP" -n "$KEYVAULT_NAME"
 az keyvault secret set --vault-name "$KEYVAULT_NAME" -n "$KEYVAULT_SECRET_NAME" --value test-value
 
 echo 'Creating a keyvault-identity and assign appropriate roles...'
 az identity create -g "$RESOURCE_GROUP" -n keyvault-identity
-IDENTITY_CLIENT_ID=$(az identity show -g "$RESOURCE_GROUP" -n keyvault-identity  --query 'clientId' -otsv)
+IDENTITY_CLIENT_ID=$(az identity show -g "$RESOURCE_GROUP" -n keyvault-identity --query 'clientId' -otsv)
 IDENTITY_PRINCIPAL_ID=$(az identity show -g "$RESOURCE_GROUP" -n keyvault-identity --query 'principalId' -otsv)
-az role assignment create --role 'Managed Identity Operator' --assignee "$AZURE_CLIENT_ID" --scope "/subscriptions/$SUBSCRIPTION_ID/resourcegroups/$RESOURCE_GROUP/providers/Microsoft.ManagedIdentity/userAssignedIdentities/keyvault-identity" || true
```

Comment:
Why are we removing these roles? This would still be required to grant the SP/cluster-identity permissions to assign the identity.

Reply:
I was assuming that the service principal / cluster-identity has `Contributor` role within the resource group, which should have the permission to operate identities inside the resource group. But we could be more explicit in the script, let me add this back.